### ISSUE TITLE
Activation Patching Tutorial Bug

### ISSUE THREAD

**Issue by jiito** [ndif-team/nnsight#502]
### **Description**

I am new to mechanistic interpretability and NNsight, so I don't quite understand what's happening here. It's likely that this behavior is described in the "Walkthrough" and I have missed it. 

However, it seems that the [activation patching tutorial's batch example](https://nnsight.net/notebooks/tutorials/causal_mediation_analysis/activation_patching/#Note:-Optimize-workflow-with-NNsight-batching) is broken when trying to patch the corrupted model's activiations, resulting in interventions that don't actually modify the model's behavior.

### **Expected Behavior**
When applying activation patches using `model.transformer.h[layer_idx].output[0][:, token_idx, :] = clean_hs[layer_idx][:, token_idx, :]`, the patched logits should differ from the corrupted logits, showing that the intervention was successful.

### **Actual Behavior**
The patched logit differences are identical to the corrupted logit differences, resulting in zero improvement values. This indicates that the interventions are not being properly registered or applied.

**Confusingly,** this assignment worked as expected in the non-batched case but seemed to perform differently in the invoker context. 

### **Steps to Reproduce**
1. Follow the activation patching tutorial
2. Use the batched approach with `tracer.invoke()` contexts
3. Apply the patch using direct assignment: `model.transformer.h[layer_idx].output[0][:, token_idx, :] = clean_hs[layer_idx][:, token_idx, :]`
4. Observe that `patched_logit_diff` equals `corrupted_logit_diff`

### **Code That Doesn't Work**
```python
# Patching corrupted run at given layer and token
with tracer.invoke(corrupted_prompt) as invoker:
    # ❌ This approach fails
    model.transformer.h[layer_idx].output[0][:, token_idx, :] = clean_hs[layer_idx][:, token_idx, :]
    
    patched_logits = model.lm_head.output
    # patched_logit_diff will equal corrupted_logit_diff
```

### **Code That Works**
```python
# Patching corrupted run at given layer and token
with tracer.invoke(corrupted_prompt) as invoker:
    # ✅ This approach works
    layer_output = model.transformer.h[layer_idx].output[0]
    layer_output[:, token_idx, :] = clean_hs[layer_idx][:, token_idx, :]
    
    patched_logits = model.lm_head.output
    # patched_logit_diff will now differ from corrupted_logit_diff
```

### **Environment**
- NNsight version: 0.4.11+
- Python version: 3.13
- OS: macOS 24.5.0

### **Additional Context**
I am new to mechanistic interpretability and NNsight, so I don't quite understand what's happening here. I'd love to learn what's going on so I can fix these in the future!

**Comment by JadenFiotto-Kaufman** [comment]
Two things conspired here:

1. **The chained-LHS write quirk** (`.output[0][:, idx, :] = ...` not propagating, while `o = .output[0]; o[:, idx, :] = ...` did) was a v0.4 nnsight artifact and is gone in v0.5+.
2. **The `[0]` indexing itself** changed shape with `transformers>=5.0` — GPT-2 block `.output` now returns the hidden-state tensor directly (not a `(hidden, ...)` tuple), so the canonical pattern is just `model.transformer.h[L].output[:, -1, :] = clean_hs` (no `[0]`).

The current pattern doc reflects this: see [`docs/patterns/activation-patching.md`](https://github.com/ndif-team/nnsight/blob/main/docs/patterns/activation-patching.md). Verified locally on current main + transformers 5.x — patching layer 7 raises P(Paris) on the corrupt prompt as expected. Closing — reopen if the new pattern doc still fails.
